lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2260,6 +2260,10 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
       originalParams.getArray(),
       [&](ParamDecl *decl) { return decl->getInterfaceType(); });
 
+  auto originalParamNames = map<SmallVector<Identifier, 8>>(
```

Comment:
This map is a bit inefficient. If all you need is to find the index, `llvm::find_if` will get you there without creating a new vector.

Reply:
Done in https://github.com/apple/swift/pull/22000/commits/aec0072e4eb43e7c59c6b73d980030413deaceea.